optuna/visualization/matplotlib/_contour.py

```diff
@@ -206,243 +176,152 @@ def get_indices(self) -> List[int]:
-    List[Union[int, float]],
     List[int],
     List[str],
     List[int],
     List[str],
-    int,
-    int,
+    List[Union[int, float]],
+    List[Union[int, float]],
 ]:
 
-    # Extract values for x, y, z axes from each trail.
     x_values = []
     y_values = []
     z_values = []
-    x_range_values = []
-    y_range_values = []
-    for trial in trials:
-        contains_x_param = x_param in trial.params
-        if contains_x_param:
-            x_range_values.append(trial.params[x_param])
-
-        contains_y_param = y_param in trial.params
-        if contains_y_param:
-            y_range_values.append(trial.params[y_param])
-
-        if not contains_x_param or not contains_y_param:
-            continue
-        x_values.append(trial.params[x_param])
-        y_values.append(trial.params[y_param])
-
-        if target is None:
-            value = trial.value
-        else:
-            value = target(trial)
-
-        if isinstance(value, int):
-            value = float(value)
-        elif not isinstance(value, float):
-            raise ValueError(
-                "Trial{} has COMPLETE state, but its target value is non-numeric.".format(
-                    trial.number
-                )
-            )
-        z_values.append(value)
+    for x_value, y_value in zip(xaxis.values, yaxis.values):
+        if x_value is not None and y_value is not None:
+            x_values.append(x_value)
+            y_values.append(y_value)
+            x_i = xaxis.indices.index(x_value)
+            y_i = yaxis.indices.index(y_value)
+            z_values.append(z_values_dict[(x_i, y_i)])
 
     # Return empty values when x or y has no value.
     if len(x_values) == 0 or len(y_values) == 0:
-        return (
-            np.array([]),
-            np.array([]),
-            np.array([]),
-            x_values,
-            y_values,
-            [],
-            [],
-            [],
-            [],
-            [],
-            [],
-            0,
-            0,
-        )
+        return np.array([]), np.array([]), np.array([]), [], [], [], [], [], []
 
     # Add dummy values for grid data calculation when a parameter has one unique value.
-    x_values_dummy = []
-    y_values_dummy = []
     if len(set(x_values)) == 1:
-        x_values_dummy = [x for x in x_indices if x not in x_values]
+        x_values_dummy = [x for x in xaxis.indices if x not in x_values]
         x_values = x_values + x_values_dummy * len(x_values)
         y_values = y_values + (y_values * len(x_values_dummy))
         z_values = z_values + (z_values * len(x_values_dummy))
     if len(set(y_values)) == 1:
-        y_values_dummy = [y for y in y_indices if y not in y_values]
+        y_values_dummy = [y for y in yaxis.indices if y not in y_values]
         y_values = y_values + y_values_dummy * len(y_values)
         x_values = x_values + (x_values * len(y_values_dummy))
         z_values = z_values + (z_values * len(y_values_dummy))
```

Comment:
In `_generate_contour_subplot`, we explicitly generate an empty plot when a param has a unique value, so I suppose the conditions are always false: can we remove both two if blocks?